kcidb/monitor/spool/__init__.py

```diff
@@ -97,29 +96,46 @@ def post(self, notification, timestamp=None):
             timestamp:      An "aware" datetime.datetime object specifying the
                             notification creation time, or None to use
                             datetime.datetime.now(datetime.timezone.utc).
+            due:            An "aware" datetime.datetime object specifying the
+                            time notification should be sent out,
+                            or None to use "timestamp".
 
         Returns:
-            True, if the notification was posted onto the spool,
-            False, if not (it was already there).
+            True, if the new notification was posted onto the spool,
+            or the existing, "unpicked" notification was updated,
+            False, if not (it existed and was "picked" at the moment).
         """
         assert isinstance(notification, Notification)
         assert timestamp is None or \
             isinstance(timestamp, datetime.datetime) and timestamp.tzinfo
         if timestamp is None:
             timestamp = datetime.datetime.now(datetime.timezone.utc)
+        message = notification.render().as_string(policy=email.policy.SMTPUTF8)
+        due = notification.message.due
+        doc = self._get_doc(notification.id)
```

Comment:
Could you move this line into the `create_or_update` closure? Nothing outside it needs it, and this way we'll protect ourselves from accidentally adding some code accessing the doc outside that function, and thus outside the transaction, which is supposed to protect data consistency.